libcudacxx/include/cuda/__memory_resource/resource_ref.h

```diff
@@ -373,6 +505,7 @@ class basic_resource_ref
                          _LIBCUDACXX_AND __properties_match<_OtherProperties...>)
   _CCCL_NODISCARD bool operator==(const basic_resource_ref<_Alloc_type, _OtherProperties...>& __rhs) const
   {
+    // BUGBUG: comparing function pointers like this can lead to false negatives:
```

Comment:
We discussed this a ton during the initial design, and did not find a better way to do it. 

The problem is that in case of nested `resource_ref` we would need a way to generically discern whether a `resource_ref` is created from an owning resource or an resource adaptor that would just want to be comparable with the original type.

I guess we could create a CPO that forwards to an existing equality comparison if existing and otherwise returns false.